### PR TITLE
remove unnecessary headers

### PR DESCRIPTION
**Pull request by kbrock**
remove unnecessary headers which may help attackers in planning further attacks.

related to https://github.com/ManageIQ/manageiq-appliance/pull/314

### CODE REVIEW

File: manageiq-operator/pkg/helpers/miq-components/httpd_conf.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ func httpdApplicationConf(applicationDomain string) string {
 Listen 8080
 # Timeout: The number of seconds before receives and sends time out.
 Timeout 120
+ServerSignature Off
+ServerTokens Prod
```
Comment (Fryguy) [inline manageiq-operator/pkg/helpers/miq-components/httpd_conf.go:24]: consistency wise, here you put it after the timeout, but above you put it after the listen.  We should be consistent with whatever we did in the manageiq-appliance-build

**Comment by kbrock** [comment]
@Fryguy made the 2 httpd.conf files similar. @bdunne let me know if you prefer a different ordering.

**Comment by lrhamann** [comment]
@kbrock @Fryguy I am testing on inframgmtinstall.apps.b-cp4mcm-23-fp1-2.cp.fyre.ibm.com/api but do not see the fix for this issue.  Did this fix get pushed downstream on podified?

I still see the Apache version in the results:
```
Larrys-MBP-2:v2.3 larryhamann$ curl -k -i -X GET -H 'Accept: application/json'  -H "X-Auth-Token: 8586ffbcc2e4f6e84ed5d7f09a0c6e31"  https://inframgmtinstall.apps.b-cp4mcm-23-fp1-2.cp.fyre.ibm.com/api/vms
HTTP/1.1 500 Internal Server Error
Date: Tue, 22 Jun 2021 19:28:33 GMT
Server: Apache/2.4.37 (Red Hat Enterprise Linux)
Content-Type: application/json; charset=utf-8
Cache-Control: no-cache
X-Request-Id: 1a3536de-db25-4a28-be72-98230760c7ae
X-Runtime: 0.026792
Content-Security-Policy: default-src 'self'; child-src 'self'; connect-src 'self'; script-src 'unsafe-eval' 'unsafe-inline' 'self'; style-src 'unsafe-inline' 'self'; report-uri /dashboard/csp_report
Strict-Transport-Security: max-age=631139040
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
X-XSS-Protection: 1; mode=block
Transfer-Encoding: chunked
Set-Cookie: 6b608f527b1eca7e2403f2ef784e0a8a=31ebbd8aa285ffbd966fd6865afc30f1; path=/; HttpOnly; Secure; SameSite=None
```

**Comment by chessbyte** [comment]
@lrhamann since it is labeled `lasker/yes?` (as opposed to `lasker/backported`), it is a candidate to be backported to the Lasker branch - which would then make it into a 2.3 downstream build

**Comment by Fryguy** [comment]
Backported to `lasker` in commit b7d60d7ae7fcd93cfac18b781e661d4cc0108e31.

```text
commit b7d60d7ae7fcd93cfac18b781e661d4cc0108e31
Author: Jason Frey <fryguy9@gmail.com>
Date:   Wed May 5 12:36:08 2021 -0400

    Merge pull request #712 from kbrock/http-server-headers
    
    remove unnecessary headers
    
    (cherry picked from commit 24e3c7db6e3c4a1790cb42a333eff079eb920b8a)
```

**Pull request merged**